control-plane/connect-inject/endpoints_controller.go

```diff
@@ -1089,92 +969,6 @@ func shouldIgnore(namespace string, denySet, allowSet mapset.Set) bool {
 	return false
 }
 
-// filterAgentPods receives meta and object information for Kubernetes resources that are being watched,
-// which in this case are Pods. It only returns true if the Pod is a Consul Client Agent Pod. It reads the labels
-// from the meta of the resource and uses the values of the "app" and "component" label to validate that
-// the Pod is a Consul Client Agent.
-func (r *EndpointsController) filterAgentPods(object client.Object) bool {
```

Comment:
This code might come back to better handle upgrades but removing it for now as I'm not sure yet how upgrades will be handled.